mne/coreg.py

```diff
@@ -1245,3 +1248,637 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+        if self._has_hpi_data and self._hpi_weight > 0:
+            head_pts.append(self._dig['hpi'])
+            mri_pts.append(self._processed_high_res_mri_points[
+                self._nearest_transformed_high_res_mri_idx_hpi])
+            weights.append(np.full(len(mri_pts[-1]), self._hpi_weight))
+        head_pts = np.concatenate(head_pts)
+        mri_pts = np.concatenate(mri_pts)
+        weights = np.concatenate(weights)
+        if n_scale_params == 0:
+            mri_pts *= self._scale  # not done in fit_matched_points
+        return head_pts, mri_pts, weights
+
+    @verbose
+    def fit_icp(self, n_iterations=20, lpa_weight=1., nasion_weight=10.,
+                rpa_weight=1., verbose=None):
+        """Find MRI scaling, translation, and rotation to match HSP.
+
+        Parameters
+        ----------
+        n_iterations : int
+            Maximum number of iterations.
+        lpa_weight : float
+            Relative weight for LPA. The default value is 1.
+        nasion_weight : float
+            Relative weight for nasion. The default value is 10.
+        rpa_weight : float
+            Relative weight for RPA. The default value is 1.
+        %(verbose)s
+
+        Returns
+        -------
+        self : Coregistration
+            The modified Coregistration object.
+        """
+        if verbose:
+            self._log_dig_mri_distance()
+        if self._scale_mode is None:
+            n_scale_params = 0
+        elif self._scale_mode == 'uniform':
+            n_scale_params = 1
+        else:
+            n_scale_params = 3
+        self._lpa_weight = lpa_weight
+        self._nasion_weight = nasion_weight
+        self._rpa_weight = rpa_weight
+
+        # Initial guess (current state)
+        est = self._parameters
+        est = est[:[6, 7, None, 9][n_scale_params]]
+
+        # Do the fits, assigning and evaluating at each step
+        for _ in range(n_iterations):
+            head_pts, mri_pts, weights = self._setup_icp(n_scale_params)
+            est = fit_matched_points(mri_pts, head_pts, scale=n_scale_params,
+                                     x0=est, out='params', weights=weights)
+            if n_scale_params == 0:
+                self._update_params(rot=est[:3], tra=est[3:6])
+            elif n_scale_params == 1:
+                est = np.array(list(est) + [est[-1]] * 2)
+                self._update_params(rot=est[:3], tra=est[3:6], sca=est[6:9])
+            else:
+                self._update_params(rot=est[:3], tra=est[3:6], sca=est[6:9])
+            angle, move, scale = self._changes
+            if angle <= self._icp_angle and move <= self._icp_distance and \
+                    all(scale <= self._icp_scale):
+                break
+        if verbose:
+            self._log_dig_mri_distance()
+        return self
+
+    def omit_hsp_points(self, distance):
+        """Exclude head shape points that are far away from the MRI head.
```

Comment:
You need to be consistent in docstirngs. Above you only said HSP. I would suggest to avoid as much as possible acronyms.